registration/app/registration/services/WindowsNotificationRegistrar.scala

```diff
@@ -22,11 +22,10 @@ class WindowsNotificationRegistrar(hubClient: NotificationHubClient)(implicit ec
       .update(regId, RawWindowsRegistration.fromMobileRegistration(registration))
       .map(hubResultToRegistrationResponse)
 
-    val channelUri = registration.deviceId
-    hubClient.registrationsByChannelUri(channelUri).flatMap {
+    hubClient.registrationsByChannelUri(oldChannelUri).flatMap {
```

Comment:
shouldn't we also check if channelUri from Registration (deviceId) is already taken by some registration in hub ? Or we can optimistically assume device will not send us existing channel in new Registration